### PR TITLE
403 review comments

### PR DESCRIPTION
**Pull request by mikehalmamoj**
* Use Instant instead of LocalDateTime
* Tidy up exception hander
* Take the user from the token rather than passing in the API
* Use kotlin-logging library

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt [hunk 1]
```diff
@@ -18,6 +18,8 @@ import org.springframework.web.client.RestClientResponseException
 import javax.persistence.EntityNotFoundException
 import javax.validation.ValidationException
 
+private val log = KotlinLogging.logger {}
```
Comment (NathanAtClarity) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt:21]: 👍

**Review by NathanAtClarity — COMMENTED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt [hunk 2]
```diff
@@ -29,7 +31,7 @@ class SendLegalMailToPrisonsApiExceptionHandler {
       .body(
         ErrorResponse(
           status = BAD_REQUEST.value(),
-          userMessage = "Failed to read the payload - ${e.message}",
+          userMessage = "Failed to read the payload}",
           developerMessage = e.message
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt:35]: Extra } ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt [hunk 3]
```diff
@@ -99,21 +101,21 @@ class SendLegalMailToPrisonsApiExceptionHandler {
       .body(
         ErrorResponse(
           status = NOT_FOUND.value(),
-          userMessage = "Not found: ${e.message}",
+          userMessage = "Not found.}",
           developerMessage = e.message
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/config/SendLegalMailToPrisonsApiExceptionHandler.kt:105]: extra } ?

**Review by timharrison-moj — APPROVED** [review]
LGTM. I need to investigate this kotlin logger.

**Pull request merged**